pandas/core/indexing.py

```diff
@@ -1547,8 +1547,7 @@ def _setitem_with_indexer(self, indexer, value):
         info_axis = self.obj._info_axis_number
 
         # maybe partial set
-        # _is_mixed_type has the side effect of consolidating in-place
-        take_split_path = self.obj._is_mixed_type
+        take_split_path = len(self.obj._mgr.blocks) > 1
```

Comment:
worth defining ```__len__``` BlockManager ?

Reply:
i think that would be ambiguous as to whether it was this or `len(mgr.items)`